File: angularapp/src/app/services/driver-request.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { DriverRequest } from '../models/driver-request.model';
import { DriverRequestDto } from '../models/driver-request-dto.model';

@Injectable({
  providedIn: 'root'
})
export class DriverRequestService {

  readonly apiUrl="https://8080-eacadeababedeffaddfeeacfcadcfdab.premiumproject.examly.io/api/driverRequest";

  constructor(private http:HttpClient) { }

  getAllDriverRequests(): Observable<DriverRequest[]> {
    return this.http.get<DriverRequest[]>(`${this.apiUrl}`);
  }
 
  getDriverRequestById(driverRequestId: number): Observable<DriverRequest> {
    return this.http.get<DriverRequest>(`${this.apiUrl}/${driverRequestId}`);
  }
 
  getDriverRequestsByUserId(userId: number): Observable<DriverRequest[]> {
    return this.http.get<DriverRequest[]>(`${this.apiUrl}/user/${userId}`);
  }
 
  getDriverRequestsByDriverId(driverId: number): Observable<DriverRequest[]> {
    return this.http.get<DriverRequest[]>(`${this.apiUrl}/drivers/${driverId}`);
  }
  
  addDriverRequest(driverRequestDto: DriverRequestDto): Observable<DriverRequest> {
    return this.http.post<DriverRequest>(this.apiUrl, driverRequestDto);
  }
 
  updateDriverRequest(driverRequestId: number, driverRequest: DriverRequest): Observable<DriverRequest> {
    return this.http.put<DriverRequest>(`${this.apiUrl}/${driverRequestId}`, driverRequest);
  }
 
  deleteDriverRequest(driverRequestId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${driverRequestId}`);
  }
}

File: springapp/src/main/java/com/examly/springapp/service/DriverRequestService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.exceptions.DuplicateDriverRequestException;
import com.examly.springapp.model.DriverRequest;
import com.examly.springapp.model.DriverRequestDTO;

import jakarta.persistence.EntityNotFoundException;

public interface DriverRequestService {
    DriverRequest addDriverRequest(DriverRequestDTO driverRequestDTO) throws Exception;

    DriverRequest getDriverRequestById(Long driverRequestId) throws EntityNotFoundException;

    List<DriverRequest> getAllDriverRequests();

    DriverRequest updateDriverRequest(Long driverRequestId, DriverRequest driverRequest) throws EntityNotFoundException;

    DriverRequest deleteDriverRequest(Long driverRequestId) throws EntityNotFoundException;

    List<DriverRequest> findDriverRequestsByUserId(Long userId);

    List<DriverRequest> findDriverRequestsByDriverId(Long driverId);

}
